contracts/BaseStrategy.sol

```diff
@@ -489,35 +492,44 @@ abstract contract BaseStrategy {
 
     /**
      * @notice
-     *  Used to change `profitFactor`. `profitFactor` is used to determine
-     *  if it's worthwhile to harvest, given gas costs. (See `harvestTrigger()`
-     *  for more details.)
+     *  Used to ensure that any significant credit a strategy has from the
+     *  vault will be automatically harvested.
      *
-     *  This may only be called by governance or the strategist.
-     * @param _profitFactor A ratio to multiply anticipated
-     * `harvest()` gas cost against.
+     *  This may only be called by governance or management.
+     * @param _creditThreshold The number of want tokens that will
+     *  automatically trigger a harvest.
      */
-    function setProfitFactor(uint256 _profitFactor) external onlyAuthorized {
-        profitFactor = _profitFactor;
-        emit UpdatedProfitFactor(_profitFactor);
+    function setCreditThreshold(uint256 _creditThreshold) external onlyVaultManagers {
+        creditThreshold = _creditThreshold;
+        emit UpdatedCreditThreshold(_creditThreshold);
     }
 
     /**
      * @notice
-     *  Sets how far the Strategy can go into loss without a harvest and report
-     *  being required.
+     *  Used to automatically trigger a harvest by our keepers. Can be
+     *  useful if gas prices are too high now, and we want to harvest
+     *  later once prices have lowered.
      *
-     *  By default this is 0, meaning any losses would cause a harvest which
-     *  will subsequently report the loss to the Vault for tracking. (See
-     *  `harvestTrigger()` for more details.)
+     *  This may only be called by governance or management.
+     * @param _forceHarvestTriggerOnce Value of true tells keepers to harvest
+     *  our strategy
+     */
+    function setForceHarvestTriggerOnce(bool _forceHarvestTriggerOnce) external onlyVaultManagers {
+        forceHarvestTriggerOnce = _forceHarvestTriggerOnce;
+        emit ForcedHarvestTrigger();
```

Comment:
I would add the `bool _forceHarvestTriggerOnce` as parameter, also check we are modifying `forceHarvestTriggerOnce` value [here](https://github.com/yearn/yearn-vaults/pull/546/files#diff-517cca2fbf046cbfefd26edb1fec816e5b2a5571fd99927dc81d6b4be60cd2a7R819) without emitting an event.

Reply:
although we technically can set this to `false` if needed (perhaps, if we accidentally set it to `true` but can't harvest for some reason), the only time practically it will be used is to set to `true` to trigger a harvest, thus I didn't include the bool in the event. does that make sense?

Reply:
In general, every time we modify an state variable, we need to emit an event including the new value on it. So, the subgraph can handle the event and get the new value without calling the contract itself directly (it takes more time/resources).

In this specific case, we are modifying `forceHarvestTriggerOnce` (state variable) [here](https://github.com/yearn/yearn-vaults/pull/546/files#diff-517cca2fbf046cbfefd26edb1fec816e5b2a5571fd99927dc81d6b4be60cd2a7R819) without emitting an event, and the new value is not included in the event. So, there is no way for the subgraph to know the current value without calling the contract when the `ForcedHarvestTrigger` is emitted. We might also call the contract to get the value in the Harvest, but these situations are what make the subgraph takes too long to sync (because of all the contract calls it does).